### PR TITLE
feat: update create-react-app base to 5 and correct tooling issues

### LINKED ISSUE #1131: Feature: webpack 5 dependency updates
### Is your feature request related to a problem? Please describe.

Some of our dependencies could use some major version / breaking changes updates:
- [css-loader](https://www.npmjs.com/package/css-loader) requires some `file-loader` and `url-loader` updates as described in: [6.0.0](https://github.com/webpack-contrib/css-loader/blob/master/CHANGELOG.md#600-2021-07-14)
- [mini-css-extract-plugin](https://www.npmjs.com/package/mini-css-extract-plugin) requires some `webpack@5` updates as described in: [2.0.0](https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/CHANGELOG.md#200-2021-06-30)
- [postcss-loader](https://www.npmjs.com/package/postcss-loader) requires some `webpack@5` updates as described in: [5.0.0](https://github.com/webpack-contrib/postcss-loader/blob/master/CHANGELOG.md#500-2021-02-02)
- [webpack-manifest-plugin](https://www.npmjs.com/package/webpack-manifest-plugin) requires some `webpack@5` updates as described in: [3.0.0](https://github.com/shellscape/webpack-manifest-plugin/releases/tag/v3.0.0)
- [webpack](https://www.npmjs.com/package/webpack) requires a lot of updates as described in: [5.0.0](https://webpack.js.org/blog/2020-10-10-webpack-5-release/)
- [webpack-dev-server](https://www.npmjs.com/package/webpack-dev-server) requires a few minor updates as described in: [4.0.0](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md#400-rc0-2021-07-19)

### Describe the solution you'd like

Update webpack to version 5, update all the dependencies, then ping a maintainer to help with the major webpack updates - doing this without the peer dependencies is gruesome if not potentially impossible and should be handled in one go!

### Describe alternatives you've considered

Switch to snowpack, which is a great idea, however, without this update, we cannot accurately tell how good/big of an improvement snowpack is.

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [x] 👷 Optimization
- [ ] 📝 Documentation Update
- [x] 🔖 Release
- [ ] 🚩 Other

## Description

This PR does a bunch of things:
- updates create-react-app to 5 and webpack tooling to 5
- simplifies babel and eslint configurations
- implements new github actions cache
- fixes static deployment and diagram artifacts
- updates lockfile to version 3
- removes polyfills
- updates webpack configuration
- updates testing configuration and adds coverage report
- adds new workbox configuration
- adds web vitals configuration
- fixes all code standards issues
- makes more workflows reusable

## Related Tickets & Documents

closes #1131 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

- fix that one remaining memory leak in index
- re-investigate https://developers.google.com/web/tools/workbox/modules
- switch to *any* other package bundler
- make workflows reusable in other open sauced repos

### CODE REVIEW

File: .github/workflows/development.yml [hunk 1]
```diff
@@ -72,19 +64,11 @@ jobs:
...
-          key: ${{ runner.os }}-node-pr-${{ hashFiles('**/npm-shrinkwrap.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-pr-
-
       - name: "🔧 install npm@latest"
         run: npm i -g npm@latest
 
       - name: "📦 install dependencies"
-        run: npm ci
+        uses: bahmutov/npm-install@v1
```
Comment (bdougie) [inline .github/workflows/development.yml:71]: Looked at the repo and this is nice. Thanks for adding.

File: src/components/Contributions.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ function Contributions({repoName, owner, viewer}) {
       .catch(e => {
         console.log(e);
       });
-  }, []);
+  }, [owner, repoName, viewer]);
```
Comment (bdougie) [inline src/components/Contributions.js:29]: Curios what was happening here?

File: src/components/Issues.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ function Issues({repoName, owner}) {
       setTotal(totalCount);
       setLoading(false);
     });
-  }, [issuesFilter]);
+  }, [issuesFilter, owner, repoName]);
```
Comment (bdougie) [inline src/components/Issues.js:40]: What does this fix?

File: src/components/Repository.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ function Repository({user, match}) {
   } = match;
   const [repository, setRepository] = useState(null);
   const [error, setError] = useState(null);
-  const [note, setNote] = useState(location.note);
+  const [note, setNote] = useState(null);
```
Comment (bdougie) [inline src/components/Repository.js:29]: Curious about this as well. Does this solve an error that was overlooked?

**Review by bdougie — COMMENTED** [review]
Left a few comments. Just curious about a few things.

File: src/components/Contributions.js (hunk 2 above)
Comment (0-vortex) [reply src/components/Contributions.js:29]: All of these are basically https://reactjs.org/docs/hooks-rules.html and https://stackoverflow.com/a/58959298/757375

File: src/components/Repository.js (hunk 4 above)
Comment (mtfoley) [reply src/components/Repository.js:29]: I’ve made this adjustment offline in the vite branch in response to some errors. but haven’t committed or pushed. I don’t know what `location.note` Referred to but being that it was the prefetch state value something “empty” makes sense to have here.

Comment (0-vortex) [reply src/components/Repository.js:29]: `location.note` was never really populated - that key doesn't exist in the location object

**Comment by 0-vortex** [comment]
@bdougie this was missing some commits :D I'll cancel the release flow and re-open if that's ok 🍕

**Pull request merged**